newm-server/src/main/kotlin/io/newm/server/features/cardano/parser/NFTSongParser.kt

```diff
@@ -1,29 +1,41 @@
 package io.newm.server.features.cardano.parser
 
-import io.ktor.utils.io.core.toByteArray
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.ktor.server.application.*
+import io.ktor.utils.io.core.*
 import io.newm.chain.grpc.LedgerAssetMetadataItem
 import io.newm.chain.grpc.NativeAsset
 import io.newm.chain.util.hexStringToAssetName
 import io.newm.server.features.cardano.model.NFTSong
-import io.newm.server.features.nftcdn.repo.NftCdnRepository
+import io.newm.server.ktx.getSecureConfigString
 import io.newm.shared.koin.inject
-import io.newm.shared.ktx.debug
 import io.newm.shared.ktx.toDurationOrNull
-import io.newm.shared.ktx.warn
 import io.newm.txbuilder.ktx.fingerprint
-import org.koin.core.parameter.parametersOf
-import org.slf4j.Logger
-import java.util.UUID
+import kotlinx.coroutines.runBlocking
+import java.util.*
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import kotlin.concurrent.getOrSet
 
-private val logger: Logger by inject { parametersOf("NFTSongParser") }
-private val nftCdnRepository: NftCdnRepository by inject()
+private val logger = KotlinLogging.logger {}
+private val environment: ApplicationEnvironment by inject()
+
+private val nftCdnSubdomain: String by lazy {
+    runBlocking { environment.getSecureConfigString("nftCdn.subdomain") }
+}
+
+private val nftCdnSecretKey: ByteArray by lazy {
+    Base64.getDecoder().decode(runBlocking { environment.getSecureConfigString("nftCdn.secretKey") })
+}
+
+private val nftCdnMacContainer = ThreadLocal<Mac>()
 
 // Used to remove numeric prefixes on legacy SickCity NFT fields ( e.g., "1. Artist Name" and "02. Song Title")
 private val legacyPrefixRegex = "^\\d+\\.\\s*".toRegex()
 
 fun List<LedgerAssetMetadataItem>.toNFTSongs(
     asset: NativeAsset,
-    isNftCdnEnabled: Boolean
+    nftCdnEnabled: Boolean
```

Comment:
This one, go ahead and revert. I told you the wrong thing to do when we rebased.